internal/runtimehandlerhooks/high_performance_hooks.go

```diff
@@ -113,3 +193,89 @@ func setCPUSLoadBalancing(c *oci.Container, enable bool, schedDomainDir string)
 
 	return nil
 }
+
+func setIRQLoadBalancing(c *oci.Container, enable bool, irqSmpAffinityFile string) error {
+	lspec := c.Spec().Linux
+	if lspec == nil ||
+		lspec.Resources == nil ||
+		lspec.Resources.CPU == nil ||
+		lspec.Resources.CPU.Cpus == "" {
+		return fmt.Errorf("failed to find the container %q CPUs", c.ID())
+	}
+
+	content, err := ioutil.ReadFile(irqSmpAffinityFile)
+	if err != nil {
+		return err
+	}
+	currentIRQSMPSetting := strings.Trim(string(content), "\n")
+
+	newIRQSMPSetting, err := UpdateIRQSmpAffinityMask(lspec.Resources.CPU.Cpus, currentIRQSMPSetting, enable)
+	if err != nil {
+		return err
+	}
+	newIRQSMPSetting += "\n"
+	if err := ioutil.WriteFile(irqSmpAffinityFile, []byte(newIRQSMPSetting), 0o644); err != nil {
+		return err
+	}
+	return nil
+}
+
+func setCPUQuota(cpuMountPoint, parentDir string, c *oci.Container, enable bool) error {
+	var rpath string
+	var err error
+	var cfsQuotaPath string
+	var parentCfsQuotaPath string
+	var cgroupManager cgmgr.CgroupManager
+
+	if strings.Contains(parentDir, ".slice") {
+		// systemd fs
+		if cgroupManager, err = cgmgr.SetCgroupManager("systemd"); err != nil {
+			return nil
+		}
+		parentPath, err := systemd.ExpandSlice(parentDir)
+		if err != nil {
+			return err
+		}
+		parentCfsQuotaPath = filepath.Join(cpuMountPoint, parentPath, "cpu.cfs_quota_us")
+		if rpath, err = cgroupManager.ContainerCgroupAbsolutePath(parentDir, c.ID()); err != nil {
+			return err
+		}
+		cfsQuotaPath = filepath.Join(cpuMountPoint, rpath, "cpu.cfs_quota_us")
```

Comment:
not blocking: I recommend to refactor this function (and the cgroupfs one) into own ones to be able to test them and reduce the complexity.

Reply:
I'm working on some cgroup common utils outside this PR.  Once that's done I will do the refactor using the common utils.